### PR TITLE
Adds the 2022 Rules Section for Behaviour

### PR DESCRIPTION
**Pull request by BryceTuppurainen**
As an idea from year-to-year to ensure that the team is aware of the rules and the rule changes that apply for their subsystems that they're working on (in this case behaviour) I've added a little details section to the end of the behaviour subsystem that states what the 2022 rules are specifically regarding penalties.

As they're implemented I can update NUbook to state (and link) where in the codebase each rule is being fulfilled so that anyone looking at this in future can immediately determine which lines to edit and what to do in order to adjust for future rule changes and we can make sure that they're all being covered easily from there.

### CODE REVIEW

**Comment by BryceTuppurainen** [comment]
https://deploy-preview-229--nubook.netlify.app/ <-- Preview

**Comment by ysims** [comment]
This is in the script tuner section of the page? It would be better to put it in a rules section or play state section.

File: src/book/02-system/02-subsystems/06-behaviour.mdx [hunk 1]
```diff
@@ -80,3 +80,26 @@ A script consists of frames (also called targets) which specify the position of
 ScriptTuner also allows for "locking" and "unlocking" servos on the robot. When a servo is "locked", its position can only be changed in ScriptTuner. When it is "unlocked", the servo can be moved physically on the robot, which will update the position value in the script. Locking a servo involves emitting a ServoCommand for the current position of the servo. Unlocking the servo involves emitting a ServoCommand with zero torque and gain.
 
 To learn how to use ScriptTuner, read the [ScriptTuner guide](/guides/main/tuning-and-running-scripts#script-tuner).
+
+<details>
+<summary>Play-States and Considerations (RoboCup 2022)</summary>
+
+Based on the rules as specified in the [RoboCup Humanoid Rules](https://github.com/RoboCup-Humanoid-TC/Rules) there are some states in the game such as when a penalty occurs in favor of the opposing team which recieves either a direct or indirect free kick, where the robots need to then act in a particular way such as standing still, or moving out of the way for the kick for the duration of the kick.
```
Comment (LachlanCourt) [inline src/book/02-system/02-subsystems/06-behaviour.mdx:87]:
```suggestion
Based on the specifications in the [RoboCup Humanoid Rules](https://github.com/RoboCup-Humanoid-TC/Rules) there are some states in the game where the robots need to then act differently to their regular behaviour, for example if the opposing team has been granted a free kick,  we might need to stand still, or move out of the way for the duration of the kick.
```

File: src/book/02-system/02-subsystems/06-behaviour.mdx [hunk 2]
```diff
@@ -80,3 +80,26 @@ A script consists of frames (also called targets) which specify the position of
...
+
+[In the revision of the rules as per March 24th 2022:](http://humanoid.robocup.org/wp-content/uploads/RC-HL-2022-Rules.pdf)
+
+### _All free kicks are taken from the place where the offence occurred, except:_
```
Comment (LachlanCourt) [inline src/book/02-system/02-subsystems/06-behaviour.mdx:91]:
```suggestion
### All free kicks are taken from the place where the offence occurred, except:
```

File: src/book/02-system/02-subsystems/06-behaviour.mdx [hunk 3]
```diff
@@ -80,3 +80,26 @@ A script consists of frames (also called targets) which specify the position of
 ScriptTuner also allows for "locking" and "unlocking" servos on the robot. When a servo is "locked", its position can only be changed in ScriptTuner. When it is "unlocked", the servo can be moved physically on the robot, which will update the position value in the script. Locking a servo involves emitting a ServoCommand for the current position of the servo. Unlocking the servo involves emitting a ServoCommand with zero torque and gain.
 
 To learn how to use ScriptTuner, read the [ScriptTuner guide](/guides/main/tuning-and-running-scripts#script-tuner).
+
+<details>
+<summary>Play-States and Considerations (RoboCup 2022)</summary>
+
+Based on the rules as specified in the [RoboCup Humanoid Rules](https://github.com/RoboCup-Humanoid-TC/Rules) there are some states in the game such as when a penalty occurs in favor of the opposing team which recieves either a direct or indirect free kick, where the robots need to then act in a particular way such as standing still, or moving out of the way for the kick for the duration of the kick.
+
+[In the revision of the rules as per March 24th 2022:](http://humanoid.robocup.org/wp-content/uploads/RC-HL-2022-Rules.pdf)
```
Comment (LachlanCourt) [inline src/book/02-system/02-subsystems/06-behaviour.mdx:89]: To match the convention of dates elsewhere in nubook
```suggestion
[In the revision of the rules as per 24th of March 2022:](http://humanoid.robocup.org/wp-content/uploads/RC-HL-2022-Rules.pdf)
```

File: src/book/02-system/02-subsystems/06-behaviour.mdx [hunk 4]
```diff
@@ -80,3 +80,26 @@ A script consists of frames (also called targets) which specify the position of
...
+
+### _All free kicks are taken from the place where the offence occurred, except:_
+
+- _Indirect free kicks to the attacking team for an offence inside the opponents' penalty area are
+  taken from the nearest point on the penalty area line which runs parallel to the goal line._
```
Comment (LachlanCourt) [inline src/book/02-system/02-subsystems/06-behaviour.mdx:94]:
```suggestion
- Indirect free kicks to the attacking team for an offence inside the opponent's penalty area, which are taken from the nearest point on the penalty area line which runs parallel to the goal line.
```

File: src/book/02-system/02-subsystems/06-behaviour.mdx [hunk 5]
```diff
@@ -80,3 +80,26 @@ A script consists of frames (also called targets) which specify the position of
...
+
+- _In the physical competition, free kicks to the defending team in their goal area may be taken
+  from anywhere in that area_
```
Comment (LachlanCourt) [inline src/book/02-system/02-subsystems/06-behaviour.mdx:97]:
```suggestion
- In the physical competition, where free kicks to the defending team in their goal area may be taken from anywhere in that area
```

File: src/book/02-system/02-subsystems/06-behaviour.mdx [hunk 6]
```diff
@@ -80,3 +80,26 @@ A script consists of frames (also called targets) which specify the position of
...
+
+### _Until the ball is in play all opponents must remain:_
```
Comment (LachlanCourt) [inline src/book/02-system/02-subsystems/06-behaviour.mdx:99]:
```suggestion
### Until the ball is in play all opponents must remain:
```

File: src/book/02-system/02-subsystems/06-behaviour.mdx [hunk 7]
```diff
@@ -80,3 +80,26 @@ A script consists of frames (also called targets) which specify the position of
...
+
+- _0.75m for KidSize and 1.5m for AdultSize from the ball until it is in play, unless they are on their own goal line between the goalposts_
```
Comment (LachlanCourt) [inline src/book/02-system/02-subsystems/06-behaviour.mdx:101]:
```suggestion
- 0.75m for KidSize and 1.5m for AdultSize from the ball until it is in play, unless they are on their own goal line between the goalposts
```

File: src/book/02-system/02-subsystems/06-behaviour.mdx [hunk 8]
```diff
@@ -80,3 +80,26 @@ A script consists of frames (also called targets) which specify the position of
...
+
+- _Outside the penalty area for free kicks inside the opponents' penalty area_
```
Comment (LachlanCourt) [inline src/book/02-system/02-subsystems/06-behaviour.mdx:103]:
```suggestion
- Outside the penalty area for free kicks inside the opponent's penalty area
```

**Review by LachlanCourt — CHANGES_REQUESTED** [review]
Please add a new section for this rather than nesting it under script tuner

File: src/book/02-system/02-subsystems/06-behaviour.mdx [hunk 9]
```diff
@@ -80,3 +80,26 @@ A script consists of frames (also called targets) which specify the position of
...
+## Subsystem-Relevant RoboCup Rules
+
+<details>
+<summary>Play-States and Penalty Considerations (RoboCup 2022)</summary>
+
+Based on the specifications in the [RoboCup Humanoid Rules](https://github.com/RoboCup-Humanoid-TC/Rules) there are some states in the game where the robots must perform a particular behaviour differently to normal play. An example of this is when a penalty occurs in favor of the opposing team, which then recieves either a direct or indirect free kick, where the robots need to then act in a particular way such as standing still, or move out of the way for the duration of the kick.
+
+[In the revision of the rules as per the 24th of March 2022:](http://humanoid.robocup.org/wp-content/uploads/RC-HL-2022-Rules.pdf)
+
+### All free kicks are taken from the place where the offence occurred, except:
```
Comment (ysims) [inline src/book/02-system/02-subsystems/06-behaviour.mdx:93]: I don't think this should be a heading

**Review by LachlanCourt — DISMISSED** [review]

**Review by ysims — APPROVED** [review]

**Pull request merged**